sdks/swift/Mentat/MentatTests/MentatTests.swift

```diff
@@ -0,0 +1,836 @@
+                [:db/add "s" :db/cardinality :db.cardinality/one]
+                [:db/add "k" :db/ident :foo/keyword]
+                [:db/add "k" :db/valueType :db.type/keyword]
+                [:db/add "k" :db/cardinality :db.cardinality/one]
+                [:db/add "u" :db/ident :foo/uuid]
+                [:db/add "u" :db/valueType :db.type/uuid]
+                [:db/add "u" :db/cardinality :db.cardinality/one]
+            ]
+            """
+            let report = try mentat.transact(transaction: schema)
+            let stringEntid = report.entidForTmpId(tmpId: "s")!
+
+            let data = """
+            [
+                [:db/add "a" :foo/boolean true]
+                [:db/add "a" :foo/long 25]
+                [:db/add "a" :foo/instant #inst "2017-01-01T11:00:00.000Z"]
+                [:db/add "a" :foo/double 11.23]
+                [:db/add "a" :foo/string "The higher we soar the smaller we appear to those who cannot fly."]
+                [:db/add "a" :foo/keyword :foo/string]
+                [:db/add "a" :foo/uuid #uuid "550e8400-e29b-41d4-a716-446655440000"]
+                [:db/add "b" :foo/boolean false]
+                [:db/add "b" :foo/ref \(stringEntid)]
+                [:db/add "b" :foo/long 50]
+                [:db/add "b" :foo/instant #inst "2018-01-01T11:00:00.000Z"]
+                [:db/add "b" :foo/double 22.46]
+                [:db/add "b" :foo/string "Silence is worse; all truths that are kept silent become poisonous."]
+                [:db/add "b" :foo/uuid #uuid "4cb3f828-752d-497a-90c9-b1fd516d5644"]
+            ]
+            """
+            return try mentat.transact(transaction: data)
+        } catch {
+            assertionFailure(error.localizedDescription)
+        }
+        return nil
+    }
+
+    func test1TransactVocabulary() {
+        do {
+            let mentat = Mentat()
+            let report = try transactSchema(mentat: mentat)
+            XCTAssertNotNil(report)
+            assert(report.txId > 0)
+        } catch {
+            assertionFailure(error.localizedDescription)
+        }
+    }
+
+    func test2TransactEntities() {
+        do {
+            let mentat = Mentat()
+            let _ = try self.transactSchema(mentat: mentat)
+            let report = try self.transactEdn(mentat: mentat)
+            XCTAssertNotNil(report)
+            assert(report.txId > 0)
+            let entid = report.entidForTmpId(tmpId: "a17592186045438")
+            assert(entid == 65566)
+        } catch {
+            assertionFailure(error.localizedDescription)
+        }
+    }
+
+    func testQueryScalar() {
+        let mentat = newStore()
+        let query = "[:find ?n . :in ?name :where [(fulltext $ :community/name ?name) [[?e ?n]]]]"
+        let expect = expectation(description: "Query is executed")
+        try! mentat.query(query: query).bind(varName: "?name", toString: "Wallingford").executeScalar(callback: { (scalarResult, error) in
```

Comment:
What happens if I write:

```
let q1 = mentat.query(query: "[:find ?x :where [?x _ _]]")
let q2 = mentat.query(query: "[:find ?x :where [_ _ ?x]]")
```

or

```
try! mentat.query(…).… {
    try! mentat.query(…)
```

Reply:
Addressing the later comment about not threading inside `run` fixes this. Both tests now pass.